src/azure-cli/azure/cli/command_modules/containerapp/_utils.py

```diff
@@ -1584,7 +1584,7 @@ def set_managed_identity(cmd, resource_group_name, containerapp_def, system_assi
         containerapp_def["identity"] = {}
         containerapp_def["identity"]["type"] = "None"
 
-    if assign_system_identity and containerapp_def["identity"]["type"].__contains__("SystemAssigned"):
+    if assign_system_identity and (containerapp_def["identity"]["type"] in "SystemAssigned"):
```

Comment:
```suggestion
    if assign_system_identity and "SystemAssigned" in containerapp_def["identity"]["type"]:
```

Reply:
Thanks. I only fix the reverse issue here https://github.com/Azure/azure-cli/pull/30365#discussion_r1845819414

I forgot that more code was affected.